### DISCUSSION TITLE
Reading a SMB .xlsx file in Mirth

### DISCUSSION THREAD

**Discussion by apple162883** [nextgenhealthcare/connect#5322] [Q&A] [answered]
I have a JAR file deployed in Mirth which helps me read SMB .txt,.csv files but I am having trouble reading a .xlsx file. I read that Mirth is not able to handle .xlsx files currently. Is there any workaround solutions for this?

**Comment by pacmano1** [comment, marked as answer]
Mirth natviely reads SMB files systems and can parse txt and csv files, so your first sentence is confhsing.

"SMB" is a file sharing protocol, so it is seems somewhat irrelevant as part of your question.

As for Excel file parsing, users have had success with Apache POI.

**Reply by pacmano1** [reply]
See https://github.com/nextgenhealthcare/connect/discussions/5429

**Comment by rjkroll** [comment]
Below is a supposed solution for using .xlsx as a Source:.  Haven't tested it myself, though:

https://forums.mirthproject.io/forum/mirth-connect/development/5565-excel-as-data-source
